### ISSUE TITLE
Feature Request: custom Common Name

### ISSUE THREAD

**Issue by dhouck** [Foxboron/sbctl#362]
A common name of just `Platform Key`, `Database Key`, or `Key Exchange Key` isnʼt very descriptive.  If you might have multiple sets of keys for some reason (multiple computers, different levels of trust, etc.), it can get confusing which keys are which.  It looks like this could be worked around with the `import-keys` option, but it would still be useful to be able to pass an option to `create-keys` that provides a more detailed name.

**Comment by servies** [comment]
Isn't that already possible by means of the config file?

**Comment by dhouck** [comment]
I donʼt think so.  If you generate keys some other way you can then use the config file to point `sbctl` at them (or put them in the directory `sbctl` expects), but if you use `sbctl` to generate keys they will have the same common name every time.

**Comment by servies** [comment]
look at the function CreateKeys(...) in backend.go.
The description per key is being used (c.Keys.PK.Description etc). If it is empty the default name is eventually being used in createKey(...)
My sbctl.conf looks like this:

keydir: /var/lib/sbctl/keys
guid: /var/lib/sbctl/GUID
files_db: /var/lib/sbctl/files.json
bundles_db: /var/lib/sbctl/bundles.json
landlock: true
db_additions:
- microsoft
keys:
  pk:
    privkey: /var/lib/sbctl/keys/PK/PK.key
    pubkey: /var/lib/sbctl/keys/PK/PK.pem
    type: file
    description: My Company Platform Key
  kek:
    privkey: /var/lib/sbctl/keys/KEK/KEK.key
    pubkey: /var/lib/sbctl/keys/KEK/KEK.pem
    type: file
    description: My Company Key Exchange Key
  db:
    privkey: /var/lib/sbctl/keys/db/db.key
    pubkey: /var/lib/sbctl/keys/db/db.pem
    type: file
    description: My Company Database Key

**Comment by Foxboron** [comment]
Ah, so I did wire this up. I probably forgot to add this to the readme.

**Comment by dhouck** [comment]
It isnʼt in `sbctl.conf(5)`; I did try to look at the code but must have missed that.  Is there also a command line option to `sbctl create-keys`?  It seems like it might be useful there instead of a config file.

**Comment by Foxboron** [comment]
> It isnʼt in sbctl.conf(5); I did try to look at the code but must have missed that. Is there also a command line option to sbctl create-keys? It seems like it might be useful there instead of a config file.

No, not yet. I need to consider how metadata around keys should be exposed as flags in the CLI.
